commcare_connect/opportunity/tasks.py

```diff
@@ -73,3 +80,44 @@ def generate_payment_export(opportunity_id: int, export_format: str):
         content = content.encode()
     default_storage.save(export_tmp_name, ContentFile(content))
     return export_tmp_name
+
+
+@celery_app.task()
+def send_notification_inactive_users(opportunity_id: int):
+    opportunity = Opportunity.objects.get(opportunity_id=opportunity_id)
+    opportunity_accesses = OpportunityAccess.objects.filter(opportunity_id=opportunity_id)
+    for access in opportunity_accesses:
+        last_user_learn_module = (
+            CompletedModule.objects.filter(user=access.user, opportunity_id=opportunity_id).order_by("date").last()
+        )
+        has_started_deliver = OpportunityClaim.objects.exists(access=access)
+        if (
+            last_user_learn_module.date <= datetime.date.today() - datetime.timedelta(days=3)
+            and not has_started_deliver
+        ):
+            message = Message(
+                usernames=[access.user.username],
+                title=gettext(f"Resume your learning journey for {opportunity.name}"),
+                body=gettext(
+                    f"You have not completed your learning for {opportunity.name}."
```

Comment:
What if they have completed all the modules but haven't yet claimed the opportunity? I think we're only checking when the last module was done and not whether they have done them all.

Reply:
has_started_deliver indicates that if the Opportunity Claim exists or not. I have also added this check in both of the if blocks. If user has a opportunity claim then we dont need to check the learn modules.

Reply:
True, but it could also be the case that the user has completed the learning modules but has not yet claimed the opportunity right?

Reply:
I added this case, and refactored the code to make testing easier. https://github.com/dimagi/commcare-connect/pull/181/commits/f629e9ed52c71c65c8e04a078a53ad67e6a89441